source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/CalculationResultClientTests.cs

```diff
@@ -12,49 +12,237 @@
 using FluentAssertions;
+using Microsoft.Extensions.Options;
+using Moq;
+using Moq.Protected;
+using NodaTime;
 using Xunit;
 using Xunit.Categories;
+using static Moq.Protected.ItExpr;
 
 namespace Energinet.DataHub.Wholesale.CalculationResults.UnitTests.Infrastructure.CalculationResultClient;
 
 [UnitTest]
 public class CalculationResultClientTests
 {
-    // TODO: LRN reintroduce test after implementation of CalculationResultClient
-    // [Theory]
-    // [InlineAutoMoqData]
-    // public async Task DeserializeJson_ShouldReturnValidObject(
-    //     Guid batchId,
-    //     string gridAreaCode,
-    //     TimeSeriesType timeSeriesType,
-    //     string energySupplierGln,
-    //     string balanceResponsiblePartyGln,
-    //     CalculationResults.Infrastructure.CalculationResultClient.CalculationResultClient sut)
-    // {
-    //     // Arrange
-    //     int expectedRowCount = 96;
-    //     var expectedFirstTimeSeriesPoint = new TimeSeriesPoint(
-    //         DateTimeOffset.Parse("2023-04-04T22:00:00.000Z"),
-    //         0.000m,
-    //         QuantityQuality.Missing);
-    //     var expectedLastTimeSeriesPoint = new TimeSeriesPoint(
-    //         DateTimeOffset.Parse("2023-04-05T08:47:41.000Z"),
-    //         1.235m,
-    //         QuantityQuality.Estimated);
-    //
-    //     var stream = EmbeddedResources.GetStream("Infrastructure.CalculationResultClient.CalculationResult.json");
-    //     using var reader = new StreamReader(stream);
-    //
-    //     // Act
-    //     var actual = await sut.GetAsync(batchId, gridAreaCode, timeSeriesType, energySupplierGln, balanceResponsiblePartyGln);
-    //
-    //     // Assert
-    //     actual.TimeSeriesType.Should().Be(timeSeriesType);
-    //     actual.TimeSeriesPoints.Length.Should().Be(expectedRowCount);
-    //     actual.TimeSeriesPoints.First().Should().Be(expectedFirstTimeSeriesPoint);
-    //     actual.TimeSeriesPoints.Last().Should().Be(expectedLastTimeSeriesPoint);
-    // }
+    private readonly string[] _someGridAreas = { "123", "456", };
+
+    private readonly DatabricksOptions _someDatabricksOptions = new()
+    {
+        DATABRICKS_WAREHOUSE_ID = "anyDatabricksId",
+        DATABRICKS_WORKSPACE_URL = "https://anyDatabricksUrl",
+        DATABRICKS_WORKSPACE_TOKEN = "myToken",
+    };
+
+    private readonly Instant _somePeriodStart = Instant.FromUtc(2021, 3, 1, 10, 15);
+    private readonly Instant _somePeriodEnd = Instant.FromUtc(2021, 3, 31, 10, 15);
+    private readonly DatabricksSqlResponse _pendingDatabricksSqlResponse = new("PENDING", null);
+
+    private readonly DatabricksSqlResponse _succeededDatabricksSqlResponse =
+        new("SUCCEEDED", TableTestHelper.CreateTableForSettlementReport());
+
+    private readonly DatabricksSqlResponse _failedDatabricksSqlResponse = new("FAILED", null);
+
+    [Theory]
+    [InlineAutoMoqData]
+    public async Task GetSettlementReportResultAsync_WhenHttpStatusCodeNotOK_ThrowsException(
```

Comment:
Should it be renamed like this? I assume that the exception type matters since you made the effort to introduce it.
```suggestion
    public async Task GetSettlementReportResultAsync_WhenHttpStatusCodeNotOK_ThrowsDatabricksSqlException(
```